### DISCUSSION TITLE
GitHub Pages for a repo with multiple subfolders

### DISCUSSION THREAD

**Discussion by PaoloJr** [community/community#58276] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Is it possible to configure a single repository, with multiple subfolders (each would be a distinct web-page) to be enabled to use GItHub Pages?

For example, a repo named "my test projects", enabled for GitHub Pages, would have three folders (project A, project B, project C). Could each of those folders get a unique URL?

The goal would be to consolidate numerous similarly-purposed repositories into one repo with multiple nested projects.
many thanks in advance 👍

**Comment by airtower-luna** [comment]
I'm not sure if it'd work with the default Jekyll build, but with a [custom deployment workflow](https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#publishing-with-a-custom-github-actions-workflow), why not? It might get a little convoluted though, you'd have to make sure to run all the build steps for the submodules (if necessary) and assemble a single deployment artifact.

**Comment by the-officialjosh** [comment, marked as answer]
Yes, it is possible to configure a single repository with multiple subfolders to be enabled for GitHub Pages, and each subfolder can have its own unique URL.

Here's how you can achieve this:

1. Enable GitHub Pages for the repository: Go to the repository settings, navigate to the **Pages** section, and choose the branch or folder that contains the content you want to publish. This will enable GitHub Pages for the repository.

2. Configure the subfolders: Each subfolder (project A, project B, project C) should contain the necessary files for their respective web pages, such as HTML, CSS, and assets.

3. Set up the URL paths: By default, GitHub Pages will use the repository name as part of the URL. However, you can customize the URL paths for each subfolder by creating an `index.md` or `index.html` file inside each subfolder. For example, you can have an `index.md` file in the project A folder that contains content specific to project A, and GitHub Pages will generate a URL like `https://<username>.github.io/my-test-projects/project-a/` for that subfolder.

By following this approach, you can consolidate multiple projects into a single repository and have each subfolder represent a distinct web page with its own unique URL.

**Note:** The generated URLs will depend on the repository name and the structure of the subfolders. Make sure to choose appropriate names and folder structures to achieve the desired URLs.

I hope this helps! Let me know if you have any further questions.

**Reply by PaoloJr** [reply]
thanks for sharing Josh! I will look into this and provide feedback soon. On the point about creating an index.md (or index.html), what should be entered there? Would you know what documentation I can refer to for completing these tasks?
many thanks in advance 👍

**Reply by the-officialjosh** [reply]
Since it is a webpage you use HTML

**Reply by PaoloJr** [reply]
thanks @the-officialjosh. Each subfolder has a HTML file (along with other static files). What should I enter in the HTML file? Is there a particular tag that GitHub Pages requires for the individual URLs?

**Reply by DanKaplanSES** [reply]
It's worth mentioning that you can also put an HTML file with any name in the root or subdir and access it too: you don't _need_ to name it `index.html` or `index.md`. 

For example, I **currently** have an HTML file at [https://github.com/DanKaplanSES/jsdom-configurable-resource-loader/blob/main/test/iframe-test.html](https://github.com/DanKaplanSES/jsdom-configurable-resource-loader/blob/049802a0a90315f4461db4a84de79ce8dc6e2c17/test/iframe-test.html). You can **currently** view this page at https://dankaplanses.github.io/jsdom-configurable-resource-loader/test/iframe-test.html

(I am emphasizing the word "currently" because I may move this file in the future, in which case the 2nd link might 404, but I still think it's helpful to show these urls.)

**Reply by ghost** [reply]
Thanks a lot, this really helped me

**Comment by PaoloJr** [comment]
Hello again @the-officialjosh, I now understand! To access each subfolder's app via browser (they each have the necessary minified files), I simply needed to append the folder-name to the URL while still using the GitHub Pages repo live-site (like you mentioned). Many thanks ✌️

**Reply by the-officialjosh** [reply]
Could you please mark that as the answer so others who need similar help can access it?

**Comment by DanatGutema** [comment]
Hello,
thank you so much @the-officialjosh, for your explanation. it was helpful for me too.
